pandas/_libs/interval.pyi

```diff
@@ -0,0 +1,96 @@
+from datetime import timedelta
+import numbers
+from typing import (
+    Generic,
+    TypeVar,
+)
+
+import numpy as np
+import numpy.typing as npt
+
+from pandas._typing import (
+    IntervalCloseness,
+    Scalar,
+)
+
+_ScalarT = TypeVar("_ScalarT", bound=Scalar)
+
+class IntervalMixin:
+    @property
+    def closed_left(self) -> bool: ...
+    @property
+    def closed_right(self) -> bool: ...
+    @property
+    def open_left(self) -> bool: ...
+    @property
+    def open_right(self) -> bool: ...
+    @property
+    def mid(self) -> float: ...
+    @property
+    def length(self) -> float: ...
```

Comment:
could be timedelta?

Reply:
It is now generic but IntervalMixin[datetime].length is annotated to return a datetime. I don't think this exception can be achieved with overloads (overlapping overloads with different return type).